enterprise/frontend/src/embedding/data-picker/DataSelector/tests/DataSourceSelector.unit.spec.tsx

```diff
@@ -0,0 +1,224 @@
+  describe("only tables are available", () => {
+    const setupOpts: SetupOpts = {
+      availableModels: "tables-only",
+    };
+
+    it("should close the picker when clicking outside", async () => {
+      setup(setupOpts);
+      expect(await screen.findByText("Orders")).toBeInTheDocument();
+      expect(screen.getByText("Sample Database")).toBeInTheDocument();
+
+      await userEvent.click(
+        screen.getByText("Click me to open or close data picker"),
+      );
+
+      expect(screen.queryByText("Orders")).not.toBeInTheDocument();
+      expect(screen.queryByText("Sample Database")).not.toBeInTheDocument();
+    });
+
+    it("should show search input when there are 10 and more tables", async () => {
+      const sampleDatabase = createSampleDatabase();
+      const DB_ID = getNextId(sampleDatabase.id);
+      setup({
+        ...setupOpts,
+        databases: [
+          sampleDatabase,
+          createMockDatabase({
+            id: DB_ID,
+            name: "Many tables Database",
+            tables: [...createNNumberOfTables(10, DB_ID)],
+          }),
+        ],
+      });
+      expect(await screen.findByText("Sample Database")).toBeInTheDocument();
+      expect(screen.getByText("Many tables Database")).toBeInTheDocument();
+
+      await userEvent.click(await screen.findByText("Many tables Database"));
+      expect(await screen.findByText("Table 1")).toBeInTheDocument();
+      expect(await screen.findByText("Table 2")).toBeInTheDocument();
+      expect(await screen.findByText("Table 3")).toBeInTheDocument();
+      expect(await screen.findByText("Table 4")).toBeInTheDocument();
+      expect(await screen.findByText("Table 5")).toBeInTheDocument();
+      expect(await screen.findByText("Table 6")).toBeInTheDocument();
+      expect(await screen.findByText("Table 7")).toBeInTheDocument();
+      expect(await screen.findByText("Table 8")).toBeInTheDocument();
+      expect(await screen.findByText("Table 9")).toBeInTheDocument();
+      expect(await screen.findByText("Table 10")).toBeInTheDocument();
+
+      const searchBox = screen.getByPlaceholderText("Find...");
+      expect(searchBox).toBeInTheDocument();
+
+      // Assert that the search query is used.
+      await userEvent.type(searchBox, "Table 1");
+      expect(await screen.findByText("Table 1")).toBeInTheDocument();
+      expect(screen.queryByText("Table 2")).not.toBeInTheDocument();
+      expect(screen.queryByText("Table 3")).not.toBeInTheDocument();
+      expect(screen.queryByText("Table 4")).not.toBeInTheDocument();
+      expect(screen.queryByText("Table 5")).not.toBeInTheDocument();
+      expect(screen.queryByText("Table 6")).not.toBeInTheDocument();
+      expect(screen.queryByText("Table 7")).not.toBeInTheDocument();
+      expect(screen.queryByText("Table 8")).not.toBeInTheDocument();
+      expect(screen.queryByText("Table 9")).not.toBeInTheDocument();
+      expect(await screen.findByText("Table 10")).toBeInTheDocument();
```

Comment:
why does Table 2 - 9 not have an await, while the above do?

Reply:
since they're using `queryBy` they don't need to be `await`ed. Anyway, we're supposed to only `await` one element to make sure everything is loaded.

Fixed by 1112ab5373e45f4d99aa34ee1693cdc232034212